File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        // string res="";
        // int count=0;
        // int n=s.size();
        // for(int i=0;i<n;i++){
        //     if(s[i]=='('){
        //         count++;
        //     }
        //     else{
        //         count--;
        //     }
        //       if((count==1&&s[i]=='('||count==0&&s[i]==')')){
        //     continue;
        // }
        // else{
        //     res=res+s[i];
        // }

        // }
        // return res;
         string res;
        int opened = 0;
        for (char c : s) {
            if (c == '(' && opened++ > 0) res += c;
            if (c == ')' && opened-- > 1) res += c;
        }
        return res;
      
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        Stack<Character>st=new Stack<>();
        StringBuilder ans=new StringBuilder();
        for (char c : s.toCharArray()){
            if(c=='('){
                if(!st.isEmpty()){
                    ans.append('(');
                }
                st.push('(');
            }else{
                st.pop();
                if(!st.isEmpty()){
                    ans.append(')');
                }
            }
        }
         return ans.toString();
       
    }
}